frontend/src/core/codemirror/language/sql.ts

```diff
@@ -172,44 +172,60 @@ export class SQLCompletionStore {
   private cache = new LRUCache<DataSourceConnection, SQLConfig>(10);
 
   getCompletionSource(connectionName: ConnectionName): SQLConfig | null {
-    const dataSourceConnections = store.get(dataSourceConnectionsAtom);
-    const connection = dataSourceConnections.connectionsMap.get(connectionName);
+    const dataConnectionsMap = store.get(dataConnectionsMapAtom);
+    const connection = dataConnectionsMap.get(connectionName);
     if (!connection) {
       return null;
     }
 
     let cacheConfig: SQLConfig | undefined = this.cache.get(connection);
     if (!cacheConfig) {
-      // Verbose version with database.schema.table
-      const schema: Record<
+      const mapping: Record<
         string,
-        Record<string, Record<string, string[]>>
+        | string[]
+        | Record<string, string[] | Record<string, Record<string, string[]>>>
       > = {};
-      const tables: Record<string, string[]> = {};
+
+      // When there is default db and schema, we can use the table name directly
+      // Otherwise, we need to use the fully qualified name
       for (const database of connection.databases) {
-        schema[database.name] = {};
-        for (const dbSchema of database.schemas) {
-          schema[database.name][dbSchema.name] = {};
-          for (const table of dbSchema.tables) {
+        // If there is only one database, it is the default
+        const isDefaultDb =
+          connection.default_database === database.name ||
+          connection.databases.length === 1;
+
+        for (const schema of database.schemas) {
+          for (const table of schema.tables) {
             const columns = table.columns.map((col) => col.name);
-            schema[database.name][dbSchema.name][table.name] = columns;
-            // Also add the table name at the top level
-            tables[table.name] = columns;
+
+            if (isDefaultDb) {
+              const schemaMap = (mapping[schema.name] ?? {}) as Record<
```

Comment:
you can put the typecheck on the LHS `const schemaMap: Record<string, string[]>;`. this might be safer than casting